scripts/build/builders/realtek.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+from enum import Enum, auto
+
+from .builder import Builder, BuilderOutput
+
+
+class RealtekBoard(Enum):
+    RTL8777G = auto()
+
+    @property
+    def BoardName(self):
+        return 'rtl8777g'
+
+
+class RealtekApp(Enum):
+    LIGHT = auto()
+    LIGHT_SWITCH = auto()
+    LOCK = auto()
+    WINDOW = auto()
+
+    @property
+    def ExampleName(self):
+        if self == RealtekApp.LIGHT:
+            return 'lighting-app'
+        elif self == RealtekApp.LIGHT_SWITCH:
+            return 'light-switch-app'
+        elif self == RealtekApp.LOCK:
+            return 'lock-app'
+        elif self == RealtekApp.WINDOW:
+            return 'window-app'
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+    @property
+    def TargetName(self):
+        if self == RealtekApp.LIGHT:
+            return 'matter-cli-ftd'
+        elif self == RealtekApp.LIGHT_SWITCH:
+            return 'matter-cli-mtd'
+        elif self == RealtekApp.LOCK:
+            return 'matter-cli-mtd'
+        elif self == RealtekApp.WINDOW:
+            return 'matter-cli-mtd'
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+    @property
+    def AppNamePrefix(self):
+        if self == RealtekApp.LIGHT:
+            return 'chip-rtl8777g-lighting-app'
+        elif self == RealtekApp.LIGHT_SWITCH:
+            return 'chip-rtl8777g-light-switch-app'
+        elif self == RealtekApp.LOCK:
+            return 'chip-rtl8777g-lock-app'
+        elif self == RealtekApp.WINDOW:
+            return 'chip-rtl8777g-window-app'
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+
+class RealtekBuilder(Builder):
+
+    def __init__(self,
+                 root,
+                 runner,
+                 board: RealtekBoard = RealtekBoard.RTL8777G,
+                 app: RealtekApp = RealtekApp.LIGHT,
+                 enable_cli: bool = False,
+                 enable_rpc: bool = False,
+                 enable_shell: bool = False):
+        super(RealtekBuilder, self).__init__(root, runner)
+        self.board = board
+        self.app = app
+        self.enable_cli = enable_cli
+        self.enable_rpc = enable_rpc
+        self.enable_shell = enable_shell
+
+        self.ot_src_dir = os.path.join(os.getcwd(), 'third_party/openthread/ot-realtek')
+
+    def CmakeBuildFlags(self):
```

Comment:
```suggestion
    def CmakeBuildFlags(self) -> str:
```

Although I wonder why we return a str since if we exec we probably want elements instead.

Reply:
Ah, the usage is to pass things in to bash. There we probably would want to use shlex to escape thing to make all paths work ... can probably be done afterwards if we find for example that getcwd contains odd characters.